lib/SILOptimizer/FunctionSignatureTransforms/ExistentialSpecializer.cpp

```diff
@@ -70,82 +70,6 @@ class ExistentialSpecializer : public SILFunctionTransform {
 };
 } // namespace
 
-/// Find concrete type from init_existential_refs/addrs.
-static bool findConcreteTypeFromInitExistential(SILValue Arg,
-                                                CanType &ConcreteType) {
-  if (auto *IER = dyn_cast<InitExistentialRefInst>(Arg)) {
-    ConcreteType = IER->getFormalConcreteType();
-    return true;
-  } else if (auto *IE = dyn_cast<InitExistentialAddrInst>(Arg)) {
-    ConcreteType = IE->getFormalConcreteType();
-    return true;
-  }
-  return false;
-}
-
-/// Find the concrete type of the existential argument. Wrapper
-/// for findInitExistential in Local.h/cpp. In future, this code
-/// can move to Local.h/cpp.
-static bool findConcreteType(ApplySite AI, int ArgIdx, CanType &ConcreteType) {
-  bool isCopied = false;
-  auto Arg = AI.getArgument(ArgIdx);
-
-  /// Ignore any unconditional cast instructions. Is it Safe? Do we need to
-  /// also add UnconditionalCheckedCastAddrInst? TODO.
-  if (auto *Instance = dyn_cast<UnconditionalCheckedCastInst>(Arg)) {
-    Arg = Instance->getOperand();
-  }
-
-  /// Return init_existential if the Arg is global_addr.
-  if (auto *GAI = dyn_cast<GlobalAddrInst>(Arg)) {
-    SILValue InitExistential =
-        findInitExistentialFromGlobalAddrAndApply(GAI, AI, ArgIdx);
-    /// If the Arg is already init_existential, return the concrete type.
-    if (InitExistential &&
-        findConcreteTypeFromInitExistential(InitExistential, ConcreteType)) {
-      return true;
```

Comment:
Does findInitExistential handle this GlobalAddr case now? It might be the case. I need to make sure the tests I wrote for them do not fail.

Reply:
`getStackInitInst` calls `findInitExistentialFromGlobalAddr`. I think `findInitExistentialFromGlobalAddrAndApply` was just a trivial. If you have any out-of-tree test cases, please try them.